### PR TITLE
v2 market(s) query: add underlying collateral and debt

### PR DESCRIPTION
**Pull request by bobthebuidlr**

### CODE REVIEW

File: contracts/red-bank/src/query.rs [hunk 1]
```diff
@@ -38,6 +38,27 @@ pub fn query_market(deps: Deps, denom: String) -> StdResult<Option<Market>> {
...
+    let collateral_underlying_amount =
+        get_underlying_liquidity_amount(market.collateral_total_scaled, &market, block_time)?;
+
+    let debt_underlying_amount =
+        get_underlying_debt_amount(market.debt_total_scaled, &market, block_time)?;
+
+    Ok(MarketResponse {
+        debt_underlying_amount,
+        collateral_underlying_amount,
+        collateralization_rate: Decimal::from_ratio(
```
Comment (piobab) [inline contracts/red-bank/src/query.rs:54]: Sorry, can we change it to `utilization_rate`? This is what we use in interest rate formula, see:
```
let mut current_utilization_rate = if !total_collateral.is_zero() {
        Decimal::from_ratio(total_debt, total_collateral)
    } else {
        Decimal::zero()
    };
   ```

File: contracts/red-bank/src/query.rs [hunk 2]
```diff
@@ -56,6 +77,43 @@ pub fn query_markets(
         .collect()
 }
 
+pub fn query_markets_v2(
```
Comment (piobab) [inline contracts/red-bank/src/query.rs:80]: Could you please implement in similar way as `query_user_collaterals_v2`? Would be good to have new way for pagination. We could avoid spinning new query if there is no new items.

File: contracts/red-bank/tests/tests/test_query.rs [hunk 3]
```diff
@@ -306,3 +311,104 @@ fn query_user_asset_debt() {
         );
     }
 }
+
+#[test]
+pub fn test_query_market_v2() {
```
Comment (piobab) [inline contracts/red-bank/tests/tests/test_query.rs:316]: Please remove `test` prefixes, just leave `query_market_v2`

File: packages/types/src/red_bank/types.rs [hunk 4]
```diff
@@ -113,3 +113,13 @@ pub struct UserPositionResponse {
     pub weighted_liquidation_threshold_collateral: Uint128,
     pub health_status: UserHealthStatus,
 }
+
+#[cw_serde]
+pub struct MarketResponse {
```
Comment (piobab) [inline packages/types/src/red_bank/types.rs:118]: Can we rename it to `MarketV2Response`? It relates for MarketV2 query. It is nice to have query and response naming similar

File: scripts/types/generated/mars-red-bank/MarsRedBank.types.ts [hunk 5]
```diff
@@ -236,7 +247,23 @@ export interface Market {
   liquidity_rate: Decimal
   reserve_factor: Decimal
 }
+export interface MarketResponse {
```
Comment (piobab) [inline scripts/types/generated/mars-red-bank/MarsRedBank.types.ts:250]: Nice that flatten works! :)

File: packages/types/src/red_bank/types.rs (hunk 4 above)
Comment (bobthebuidlr) [reply packages/types/src/red_bank/types.rs]: Done, renamed it

File: packages/types/src/red_bank/types.rs [hunk 6]
```diff
@@ -112,3 +114,15 @@ pub struct UserPositionResponse {
     pub weighted_liquidation_threshold_collateral: Uint128,
     pub health_status: UserHealthStatus,
 }
+
+pub type PaginatedMarketV2Response = PaginationResponse<MarketV2Response>;
+
+#[cw_serde]
+pub struct MarketV2Response {
+    pub collateral_underlying_amount: Uint128,
```
[the remark this answers is not in the document]
Comment (bobthebuidlr) [reply packages/types/src/red_bank/types.rs:122]: Yeah I agree with the renaming to `collateral_total_amount`. I don't feel like it's necessary to create a `MarketV2` struct though

**Review by piobab — APPROVED** [review]
LGTM!

**Pull request merged**